### ISSUE TITLE
Top-level date/time ReQL terms

### ISSUE THREAD

**Issue by thelinuxlich** [rethinkdb/rethinkdb#5387]
So we can do easy date/time manipulation, they should return seconds:

```
r.now().sub(r.month()) // now - 2592000 seconds
r.now().sub(r.week()) // now - 604800 seconds
r.now().sub(r.day()) // now - 86400 seconds
r.now().add(r.hour(2)) // now + 7200 seconds
r.now().add(r.hours(2)) // r.hour() alias
r.now().add(r.minutes(30)) // now + 1800
```

**Comment by eduardonunesp** [comment]
+1

**Comment by tatsujin1** [comment]
I'd argue that since `r.now() + 100` returns a timestamp these should do as well.

**Comment by thelinuxlich** [comment]
Yeah, i explained it badly, only the helpers should return seconds

**Comment by tatsujin1** [comment]
Oh I see. But then I have a follow-up question, hire many seconds is a month? 😃

**Comment by encryptio** [comment]
Or even, how many seconds in a day? Daylight savings time changes make that a surprisingly difficult question when you want to do arithmetic on times. Adding a "day" to a timestamp and having the time-of-day change would be what adding 86400 seconds would do across daylight savings time changes, which could be completely incorrect behavior in some use cases, and exporting constants like this makes it easy to mistake "this time tomorrow" with "add one day to this time".

[Go's time library](https://golang.org/pkg/time/#Duration) decides to only give `Duration` constants up to `Hour`, which is arguably less useful but also avoids some confusion.

**Comment by thelinuxlich** [comment]
I think this is similar to .sample() current state, we shouldn't worry too much about purity, it's not like RethinkDB should bundle Moment.js to get daylight savings right, that's userland.

**Comment by tatsujin1** [comment]
Yeah, even a minute is uncertain if you take leap seconds into account, but I didn't want to step in that pile of poo...

**Comment by marshall007** [comment]
Related to #4187. These would play nicely together.

**Comment by VeXocide** [comment]
I vehemently disagree that we shouldn't worry about purity, quite the contrary, we should such that you don't have to. If treating a day as 86400 seconds is sufficient for you storing that number in a constant `day` and simply using `r.now().sub(day)` is just about as much effort as using `r.now().sub(r.day())`. However, someone expecting us to do the right, but tricky, thing would be sorely mistaken.

If we add this feature, given that it's possible to do right to begin with, we should do so and not cut corners.

**Comment by VeXocide** [comment]
For the record, the Boost.DateTime library that we use also limits duration constants to an hour, thus this would be near impossible to get right indeed.

**Comment by kwando** [comment]
I do not like this proposal. Like @VeXocide said, it is trivial to use application level constants to achieve the same result.

**Comment by chrisvariety** [comment]
I agree, if this isn't done _right_ (e.g. with a C-level equivalent of moment.js), there's no point in doing this at all.

**Comment by thelinuxlich** [comment]
People don't bring this on the Rails community for helpers like 1.days(), but ok, let's stick with what is strictly right

**Comment by chrisvariety** [comment]
@thelinuxlich as far as I can tell, rails does this correctly without naive constants. https://github.com/rails/rails/blob/91db479ed4a03ca7e8a10e5c976782a1537130e0/activesupport/lib/active_support/core_ext/time/calculations.rb#L78-L138

**Comment by danielmewes** [comment]
I think having these terms return seconds is probably not a good idea, but that doesn't mean that it has to be impossible to do arithmetic with them.

Theoretically we could add a `duration` type that's represented in terms of months, days, hours, minutes and/or seconds. We could then support adding a duration to a datetime in a way that uses the correct length of the month, day or hour.
The semantics of this might not be completely obvious though, and it could be a bit of a pain to implement.

I'd like to keep this open, but I'll move it to `backlog` since I think it's not something that we want to do quickly and badly.
